pkgs/by-name/pi/pihole-web/package.nix

```
@@ -0,0 +1,56 @@
+{ stdenv, lib, fetchFromGitHub, php, pihole, pihole-ftl, procps }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "pihole-web";
+  version = "5.21";
+
+  src = fetchFromGitHub {
+    owner = "pi-hole";
+    repo = "web";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-+3Dl+SnV4XIMxk9+EZT+s1/43zJXq/QHYwjRSum8gdE=";
+  };
+
+  propagatedBuildInputs = [
+    pihole
+    pihole-ftl
+    procps
+  ];
+
+  postPatch = ''
+    # Redirect stderr to stdout so that errors appear on the Update Gravity page
+    substituteInPlace scripts/pi-hole/php/gravity.sh.php \
+      --replace 'sudo pihole -g' '${pihole}/bin/pihole -g 2>&1'
+
+    substituteInPlace settings.php \
+      --replace '/usr/bin/pihole-FTL' ${pihole-ftl}/bin/pihole-FTL
+
+    substituteInPlace scripts/pi-hole/php/func.php \
+      --replace 'pidof pihole-FTL' '${procps}/bin/pidof pihole-FTL' \
+      --replace "ps -p" "${procps}/bin/ps -p" \
+      --replace 'sudo pihole' ${pihole}/bin/pihole
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share
+    cp -r -t $out/share *.php img/ scripts/ style/
+
+    mkdir -p $out/share/doc/$name/
+    cp README.md $out/share/doc/$name/
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = ''
```

```suggestion
    longDescription = ''
```

And provide a short `description` in only one line.